### PR TITLE
Feature/fix sonar issues

### LINKED ISSUE #16: Fix Sonar issues

### PR DESCRIPTION
**Pull request by KurnosovaElena**

### CODE REVIEW

**Review by MaliyOna — APPROVED** [review]

File: Management_System/BLL/Services/Implementation/UserBoardService.cs [hunk 1]
```diff
@@ -9,60 +10,62 @@ namespace BLL.Services.Implementation;
...
         return userBoardDTO;
     }
 
-    public async Task<UserBoardDTO> GetByUserIdAndBoardIdAsync(Guid userId, Guid boardId, CancellationToken cancellationToken)
+    public async Task<UserBoardDto> GetByUserIdAndBoardIdAsync(Guid userId, Guid boardId, CancellationToken cancellationToken)
     {
-        var userBoardEntity = await repository.GetByUserIdAndBoardIdAsync(userId, boardId, cancellationToken);
-        return mapper.Map<UserBoardDTO>(userBoardEntity);
+        var userBoardEntity = await repository.GetByUserIdAndBoardIdAsync(userId, boardId, cancellationToken) 
+            ?? throw new NotFoundException("No user board found");
```
Comment (5Ka-me) [inline Management_System/BLL/Services/Implementation/UserBoardService.cs]: Please provide IDs too

File: Management_System/BLL/Services/Implementation/UserBoardService.cs [hunk 2]
```diff
@@ -9,60 +10,62 @@ namespace BLL.Services.Implementation;
...
 
         return usersBoardsDTO;
     }
 
-    public async Task<UserBoardDTO> Update(Guid userId, Guid boardId, CreateUserBoardDTO entity, CancellationToken cancellationToken)
+    public async Task<UserBoardDto> Update(Guid userId, Guid boardId, CreateUserBoardDto entity, CancellationToken cancellationToken)
     {
         var userBoard = await repository.GetByUserIdAndBoardIdAsync(userId, boardId, cancellationToken)
-            ?? throw new Exception("UserBoard not found");
+            ?? throw new NotFoundException("UserBoard not found");
```
Comment (5Ka-me) [inline Management_System/BLL/Services/Implementation/UserBoardService.cs]: Do the same everywhere with providing IDs

File: Management_System/BLL/Services/Implementation/UserTaskService.cs [hunk 3]
```diff
@@ -9,60 +10,62 @@ namespace BLL.Services.Implementation;
...
         return userTaskDTO;
     }
 
-    public async Task<UserTaskDTO> GetByUserIdAndTaskIdAsync(Guid userId, Guid taskId, CancellationToken cancellationToken)
+    public async Task<UserTaskDto> GetByUserIdAndTaskIdAsync(Guid userId, Guid taskId, CancellationToken cancellationToken)
     {
         var userTaskEntity = await repository.GetByUserIdAndTaskIdAsync(userId, taskId, cancellationToken);
-        return mapper.Map<UserTaskDTO>(userTaskEntity);
+
+        return userTaskEntity == null
```
Comment (5Ka-me) [inline Management_System/BLL/Services/Implementation/UserTaskService.cs]: Try another way to return value. Use `??`

File: Management_System/DAL/Context/DataGenerator.cs [hunk 4]
```diff
@@ -48,28 +48,25 @@ private static void GenerateBogusLabelsData()
         Labels.AddRange(generatedLabels);
     }
 
-    private static List<TaskStatusEntity> GetBogusStatusData(Guid boardId)
+    private static void GetBogusStatusData(Guid boardId)
```
Comment (5Ka-me) [inline Management_System/DAL/Context/DataGenerator.cs]: Naming

File: Management_System/DAL/Context/DataGenerator.cs [hunk 5]
```diff
@@ -48,28 +48,25 @@ private static void GenerateBogusLabelsData()
...
     {
         var statusGenerator = GetTaskStatusFaker(boardId);
         var generatedStatus = statusGenerator.Generate(NumberOfBoards);
         TaskStatus.AddRange(generatedStatus);
-        return generatedStatus;
     }
 
-    private static List<SubtaskEntity> GetBogusSubtasksData(Guid taskId)
+    private static void GetBogusSubtasksData(Guid taskId)
```
Comment (5Ka-me) [inline Management_System/DAL/Context/DataGenerator.cs]: Naming

File: Management_System/DAL/Context/DataGenerator.cs [hunk 6]
```diff
@@ -48,28 +48,25 @@ private static void GenerateBogusLabelsData()
...
     {
         var subtasksGenerator = GetSubtasksFaker(taskId);
         var generatedSubtasks = subtasksGenerator.Generate(NumberOfTasks);
         Subtasks.AddRange(generatedSubtasks);
-        return generatedSubtasks;
     }
 
-    private static List<TaskEntity> GetBogusTasksData(Guid userId)
+    private static void GetBogusTasksData(Guid userId)
```
Comment (5Ka-me) [inline Management_System/DAL/Context/DataGenerator.cs]: Naming

File: Management_System/DAL/Entities/Entity.cs [hunk 7]
```diff
@@ -1,6 +1,6 @@
 ﻿namespace DAL.Entities;
 
-public abstract class IEntity
+public abstract class Entity
```
Comment (5Ka-me) [inline Management_System/DAL/Entities/Entity.cs]: Better to name it like `BaseEntity` or `EntityWithId`

File: Management_System/Management_System/Program.cs [hunk 8]
```diff
@@ -25,4 +25,4 @@
 
 app.MapControllers();
 
-app.Run();
\ No newline at end of file
+await app.RunAsync();
```
Comment (5Ka-me) [inline Management_System/Management_System/Program.cs:28]: Why did you do this?

**Review by 5Ka-me — CHANGES_REQUESTED** [review]

File: Management_System/Management_System/Program.cs (hunk 8 above)
Comment (KurnosovaElena) [reply Management_System/Management_System/Program.cs:28]: I had reliability issue in sonar, that included those changes

**Review by 5Ka-me — APPROVED** [review]

File: Management_System/Management_System/Program.cs (hunk 8 above)
Comment (5Ka-me) [reply Management_System/Management_System/Program.cs:28]: OK, but do you know what this for?

**Pull request merged**